### ISSUE TITLE
Questions about masks for trainning final-fcn model

### ISSUE THREAD

**Issue by kevinlee9** [ascust/BDWSS#2]
Hi @ascust, when I looked through the code, I found that voc masks for training Final-FCN, are fusions of outputs of Web-FCN and Inital-SEC **without CRF**,  Would it be better to use outputs of Web-FCN and Inital-SEC after CRF as training labels? 

And where is the post step performed  in the ablation study table, on masks generated by Final- FCN or Web-FCN?  Are all evaluations  in the ablation study table based on masks without using Final-FCN? thanks.

**Comment by ascust** [comment]
Yes the final masks are the fusion of outputs of Web-FCN and Inital-SEC without CRF. The reason why I did not use CRF for initial-SEC is since the initial-SEC is trained with complex images, it is difficult to get accurate boundaries even with CRF. On the contrary, the reason why I used CRF for web images because those images are simple and easy to find good boundaries. 

So long story short, I think both init-SEC and web-FCN have their own roles. The init-SEC is trained with multi-class complex images, so it is good at distinguishing classes. The web-FCN is trained with simple web images, so it is good at finding boundaries. That is also the reason why I have to merge the two masks together. 

For the second question, the post step in the ablation study means the one performed on the final prediction including multi-scale inference and CRF. In the ablation study, the Final-FCN is the necessary step. Because my idea in this paper is like, since we do not have pixel-level labels in the weakly supervised setting, we want to estimate the masks in some way and use them as proxy ground truth to train a model as in a "fully supervised setting". So our goal is to estimate masks for the training images more accurately. The final-FCN is also performed as in a "fully supervised setting" with our estimated masks.

I hope I have explained it clearly.

**Comment by kevinlee9** [comment]
@ascust  Yes, I figured out through your detailed explanation:smile:.  So the score 49.3(IoU of Initial-SEC in ablation study) is got by following steps?
1. train init-SEC using voc-aug training set
2. use trained init-SEC to generate masks of voc-aug training set(No CRF)
3. train Final-FCN use masks generated in step2 as proxy ground truth
4. generate masks of voc validation set using trained Final-FCN and get corresponding IoU(No mutil-scale inference and CRF)

**Comment by ascust** [comment]
Yes. Exactly.
